libraries/consul_config.rb

```diff
@@ -20,10 +20,13 @@ class ConsulConfig < Chef::Resource
       attribute(:path, kind_of: String, name_attribute: true)
       # @!attribute owner
       # @return [String]
-      attribute(:owner, kind_of: String, default: 'consul')
+      attribute(:owner, kind_of: String, default: lazy { node['consul']['config']['owner'] })
```

Comment:
tl;dr: none at all. I've pushed a new commit to change it back and test a different way.

It took me a few minutes to understand why I did it this way. It's because I had already started updating [the Rspec](https://github.com/kingpong/consul-cookbook/blob/a5e2516354a5dc735999e761399efcdc1f314187/test/spec/libraries/consul_config_spec.rb#L16) based on my original suggestion in #322. I changed part of the implementation but forgot to change the other part since it didn't break the tests.  Anyway, it's reverted now and I'm testing the changes by using the resource property instead of an attribute.